File: pdlearning/src/web-apps/src/libraries/domain-api/src/form/models/form-with-questions.model.ts
import { FormModel, IFormModel } from './form.model';
import { FormQuestionModel, IFormQuestionModel } from './form-question.model';

import { FormAnswerModel } from './form-answer.model';
import { FormSection } from '../../form-section/models/form-section';

export interface IFormWithQuestionsModel {
  form: IFormModel;
  formQuestions: IFormQuestionModel[];
  formSections: FormSection[];
}
export class FormWithQuestionsModel implements IFormWithQuestionsModel {
  public form: FormModel = new FormModel();
  public formQuestions: FormQuestionModel[] = [];
  public formSections: FormSection[] = [];

  constructor(data?: IFormWithQuestionsModel) {
    if (data != null) {
      this.form = new FormModel(data.form);
      this.formQuestions = data.formQuestions.map(_ => new FormQuestionModel(_));
      this.formSections = data.formSections.map(_ => new FormSection(_));
    }
  }
}

export interface FormQuestionWithAnswerModel {
  formQuestion: FormQuestionModel[];
  formAnswer: FormAnswerModel;
}

File: pdlearning/src/microservices/form-microservice/Api/Application/Models/FormWithQuestionsModel.cs
using System.Collections.Generic;
using System.Linq;
using System.Web;
using Microservice.Form.Domain.Entities;
using FormEntity = Microservice.Form.Domain.Entities.Form;

namespace Microservice.Form.Application.Models
{
    public class FormWithQuestionsModel
    {
        public FormWithQuestionsModel()
        {
            // Do not remove this constructor. It is being used by JsonSerializer.Deserialize
        }

        public FormWithQuestionsModel(FormEntity formEntity, IEnumerable<FormQuestion> formQuestionEntities, IEnumerable<FormSection> formSectionEntities, bool canUnpublishStandalone)
        {
            Form = new FormModel(formEntity, canUnpublishStandalone);
            FormQuestions = formQuestionEntities.Select(question =>
            {
                var questionModel = new FormQuestionModel(question);
                questionModel.QuestionTitle = HttpUtility.HtmlDecode(questionModel.QuestionTitle);
                return questionModel;
            });
            FormSections = formSectionEntities.Select(section => new FormSectionModel(section));
        }

        public FormModel Form { get; set; }

        public IEnumerable<FormQuestionModel> FormQuestions { get; set; }

        public IEnumerable<FormSectionModel> FormSections { get; set; }
    }
}
